### PR TITLE
Add configurable Adam parameters, set beta2 to 0.95

### PR DESCRIPTION
**Pull request by gobbleturk**
This change:
* Allows users to overwrite default Adam optimizer parameters
* Sets the Maxtext default beta2 to a much stabler (less training loss spikes) value of 0.95 instead of default 0.999

Before this change 17B model with default beta2=0.999
Pre-beta-instability

After this change 17B model with beta2=0.95
post-beta-stability

### CODE REVIEW

**Review by rwitten — APPROVED** [review]
nice!

**Comment by gobbleturk** [comment]
This change:

* Allows users to overwrite default Adam optimizer parameters
* Sets the Maxtext default beta2 to a much stabler (less training loss spikes) value of 0.95 instead of default 0.999

Before this change 17B model with default beta2=0.999
Pre-beta-instability

After this change 17B model with beta2=0.95
post-beta-stability
